Userland/Applications/Mail/MailWidget.cpp

```diff
@@ -0,0 +1,480 @@
+                    return c != '\r';
+                });
+
+                builder.append(current_line);
+
+                bool consumed_end_of_line = lexer.consume_specific("\r\n");
+                VERIFY(consumed_end_of_line);
+
+                // If CRLF are immediately followed by WSP (which is either ' ' or '\t'), then it is not the end of the header and is instead just a wrap.
+                // If it's not followed by WSP, then it is the end of the header.
+                // https://datatracker.ietf.org/doc/html/rfc2822#section-2.2.3
+                if (lexer.is_eof() || (lexer.peek() != ' ' && lexer.peek() != '\t'))
+                    break;
+            }
+
+            return builder.to_string();
+        };
+
+        auto& subject_iterator_value = subject_iterator->get<1>().value();
+        auto subject_index = subject_iterator_value.find("Subject:");
+        VERIFY(subject_index.has_value());
+        auto potential_subject = subject_iterator_value.substring(subject_index.value());
+        auto subject_parts = potential_subject.split_limit(':', 2);
+        auto subject = parse_and_unfold(subject_parts.last());
+
+        if (subject.is_empty())
+            subject = "(no subject)";
+
+        auto& from_iterator_value = from_iterator->get<1>().value();
+        auto from_index = from_iterator_value.find("From:");
+        VERIFY(from_index.has_value());
+        auto potential_from = from_iterator_value.substring(from_index.value());
+        auto from_parts = potential_from.split_limit(':', 2);
+        auto from = parse_and_unfold(from_parts.last());
+
+        InboxEntry inbox_entry { from, subject };
+
+        active_inbox_entries.append(inbox_entry);
+    }
+
+    m_individual_mailbox_view->set_model(InboxModel::create(active_inbox_entries));
+    m_individual_mailbox_view->on_selection_change = [this] {
+        selected_email_to_load();
+    };
+}
+
+void MailWidget::selected_email_to_load()
+{
+    auto const& index = m_individual_mailbox_view->selection().first();
+
+    if (!index.is_valid())
+        return;
+
+    // IMAP is 1-based.
+    int id_of_email_to_load = index.row() + 1;
+
+    auto fetch_command = IMAP::FetchCommand {
+        .sequence_set = { { id_of_email_to_load, id_of_email_to_load } },
+        .data_items = {
+            IMAP::FetchCommand::DataItem {
+                .type = IMAP::FetchCommand::DataItemType::BodyStructure,
+            },
+        },
+    };
+
+    auto fetch_response = m_imap_client->fetch(fetch_command, false)->await().release_value();
+
+    if (fetch_response.status() != IMAP::ResponseStatus::OK) {
+        dbgln("Failed to retrieve the body structure of the selected e-mail. The server says: '{}'", fetch_response.response_text());
+        GUI::MessageBox::show_error(window(), String::formatted("Failed to retrieve the selected e-mail. The server says: '{}'", fetch_response.response_text()));
+        return;
+    }
+
+    Vector<int> selected_alternative_position;
```

Comment:
This should probably use an unsigned type if it should never contain negative values.

Reply:
The IMAP library currently uses `int` instead of `size_t`, so I went with `int` to avoid casting.
Should I change the IMAP library to use unsigned types in this PR?